maskrcnn_benchmark/structures/boxlist_ops.py

```diff
@@ -26,6 +26,8 @@ def boxlist_nms(boxlist, nms_thresh, max_proposals=-1, score_field="scores"):
     score = boxlist.get_field(score_field)
     keep = _box_nms(boxes, score, nms_thresh)
     if max_proposals > 0:
+        #protect against not enough keep elements
+        max_proposals = min(max_proposals, len(keep))
```

Comment:
this is not necessary. Indeed, even if `max_proposals` is larger than `keep`, the slicing will just take all elements.

Reply:
thanks for your kindly response.
I removed unnecessary lines.